ydb/core/kqp/executer_actor/kqp_partitioned_executer.cpp

```diff
@@ -62,420 +90,805 @@ class TKqpPartitionedExecuter : public TActorBootstrapped<TKqpPartitionedExecute
-        Executers[partitionIdx] = exId;
-
-        LOG_D("Created new KQP executer from Partitioned: " << exId << ", isRetry = "
-            << isRetry << ", partitionIdx = " << partitionIdx);
-
-        auto ev = std::make_unique<TEvTxUserProxy::TEvProposeKqpTransaction>(exId);
-        Send(MakeTxProxyID(), ev.release());
-    }
-
-    void Abort() {
-        SendAbortToActors();
-        Become(&TKqpPartitionedExecuter::AbortState);
-    }
-
-    void SendAbortToActors() {
-        for (size_t i = 0; i < Executers.size(); ++i) {
-            if (ExecutersResponses[i] == EExecuterResponse::SUCCESS) {
-                ExecutersResponses[i] = EExecuterResponse::NONE;
-            }
-
-            if (ExecutersResponses[i] != EExecuterResponse::ERROR) {
-                auto abortEv = TEvKqp::TEvAbortExecution::Aborted("Aborted by Partitioned Executer");
-                Send(Executers[i], abortEv.Release());
-            }
-
-            Send(BufferActors[i], new TEvKqpBuffer::TEvTerminate{});
-        }
+        Abort();
     }
 
     STFUNC(ExecuteState) {
         try {
             switch (ev->GetTypeRewrite()) {
                 hFunc(TEvKqpExecuter::TEvTxResponse, HandleExecute);
-                hFunc(TEvKqp::TEvAbortExecution, HandleAbort);
+                hFunc(TEvKqpExecuter::TEvTxDelayedExecution, HandleExecute)
+                hFunc(TEvKqp::TEvAbortExecution, HandlePrepare);
                 hFunc(TEvKqpBuffer::TEvError, HandleExecute);
             default:
                 AFL_ENSURE(false)("unknown message", ev->GetTypeRewrite());
             }
         } catch (...) {
             RuntimeError(
                 Ydb::StatusIds::INTERNAL_ERROR,
-                NYql::TIssues({NYql::TIssue("RuntimeError: ExecuteState.")}));
+                NYql::TIssues({NYql::TIssue(TStringBuilder()
+                    << "from state handler = " << CurrentStateFuncName())}));
         }
-        return;
     }
 
     void HandleExecute(TEvKqpExecuter::TEvTxResponse::TPtr& ev) {
         auto* response = ev->Get()->Record.MutableResponse();
+
+        auto it = ExecuterToPartition.find(ev->Sender);
+        if (it == ExecuterToPartition.end()) {
+            PE_LOG_D("Got TEvKqpExecuter::TEvTxResponse from unknown actor with Id = " << ev->Sender
+                << ", status = " << response->GetStatus() << ", ignore");
+            return;
+        }
+
+        PE_LOG_I("Got TEvKqpExecuter::TEvTxResponse from ActorId = " << ev->Sender
+            << ", status = " << response->GetStatus());
+
+        auto [_, partInfo] = *it;
+        AbortBuffer(partInfo->BufferId);
+        ForgetExecuterAndBuffer(partInfo);
+
         switch (response->GetStatus()) {
             case Ydb::StatusIds::SUCCESS:
-                OnSuccessResponse(GetExecuterIdx(ev->Sender));
-                break;
+                partInfo->RetryDelayMs = BatchOperationSettings.StartRetryDelayMs;
+                partInfo->LimitSize = BatchOperationSettings.MaxBatchSize;
```

Comment:
I think it will be better to do something like that
```
partInfo->LimitSize = max (partInfo->LimitSize * 2, BatchOperationSettings.MaxBatchSize)
```